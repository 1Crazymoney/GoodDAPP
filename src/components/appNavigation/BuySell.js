import React from 'react'
import { View, Text } from 'react-native'
import { createStackNavigator } from './stackNavigation'

const BuySell = props => {
  return (
    <View>
      <Text>BuySell</Text>
    </View>
  )
}

export default createStackNavigator({ BuySell })
