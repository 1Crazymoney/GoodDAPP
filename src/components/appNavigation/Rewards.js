import React from 'react'
import { View, Text } from 'react-native'
import { Link } from '@react-navigation/web'
import { createStackNavigator, PushButton } from './stackNavigation'
class InternalReward2 extends React.Component {
  // This is one way of setting params for screen
  // More useful if the params are the same for all instances of this component regarding how is included in the stack
  static navigationOptions = {
    title: 'Internal Reward Screen 2'
  }

  render() {
    return (
      <View>
        <Text>Internal Reward Screen 2 </Text>
      </View>
    )
  }
}

const InternalReward1 = props => (
  <View>
    <Text>InternalReward Screen 1</Text>
    <PushButton routeName={'InternalReward2'} navigationConfig={props.screenProps}>
      InternalReward 2
    </PushButton>
  </View>
)

const Rewards = props => {
  console.log('Rewards props', props)
  return (
    <View>
      <Text>Rewards Screen</Text>
      <PushButton routeName={'InternalReward1'} navigationConfig={props.screenProps}>
        InternalReward 1
      </PushButton>
    </View>
  )
}

export default createStackNavigator({
  Rewards,
  InternalReward1: {
    screen: InternalReward1,
    // This is one way of setting params for screen.
    // More useful if the params may change for the same component
    navigationOptions: { title: 'Internal Reward Title 1' }
  },
  InternalReward2
})
