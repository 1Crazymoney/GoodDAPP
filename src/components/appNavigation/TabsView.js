//@flow
import React from 'react'
import { View, Text, StyleSheet, Image } from 'react-native'

type TabButtonProps = {
  text: string,
  routeName: string,
  icon: string,
  goTo: (routeKey: string) => void
}

const TabButton = (props: TabButtonProps) => {
  return (
    <View style={styles.tabButton} onClick={() => props.goTo(props.routeName)}>
      <Image source={props.icon} style={styles.tabIcon} />
      <Text style={styles.tabButtonText}>{props.text}</Text>
    </View>
  )
}

type TabViewProps = {
  routes: { [string]: any },
  goTo: (routeKey: string) => void
}

const TabsView = (props: TabViewProps) => (
  <View style={styles.tabView}>
    {Object.keys(props.routes)
      .filter(routeKey => props.routes[routeKey].display !== false)
      .map(routeKey => (
        <TabButton
          key={routeKey}
          routeName={routeKey}
          text={routeKey}
          goTo={props.goTo}
          icon={props.routes[routeKey].icon}
        />
      ))}
  </View>
)

const styles = StyleSheet.create({
  tabButtonText: {
    color: '#d2d2d2',
    textAlign: 'center'
  },
  tabButton: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    cursor: 'pointer'
  },
  tabView: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    height: '60px',
    backgroundColor: '#909090'
  },
  tabIcon: { width: 40, flexGrow: 2, flexBasis: 0 }
})

export default TabsView
