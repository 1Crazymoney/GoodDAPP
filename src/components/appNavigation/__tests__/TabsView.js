import React from 'react'
import TabsView from '../TabsView'
import Rewards from '../Rewards'
import BuySell from '../BuySell'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('TabsView', () => {
  const routes = {
    Rewards: {
      screen: Rewards
    },
    BuySell: {
      screen: BuySell
    }
  }

  it('renders without errors', () => {
    const tree = renderer.create(<TabsView routes={routes} />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it.only('matches snapshot', () => {
    const component = renderer.create(<TabsView routes={routes} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
