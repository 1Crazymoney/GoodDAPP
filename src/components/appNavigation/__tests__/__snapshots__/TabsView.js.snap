// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsView matches snapshot 1`] = `
<div
  style={
    Object {
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "horizontal",
      "WebkitBoxPack": "justify",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "row",
      "WebkitFlexShrink": 0,
      "WebkitJustifyContent": "space-around",
      "alignItems": "stretch",
      "backgroundColor": "rgba(144,144,144,1.00)",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "color": "rgba(210,210,210,1.00)",
      "display": "flex",
      "flexBasis": "auto",
      "flexDirection": "row",
      "flexShrink": 0,
      "height": "60px",
      "justifyContent": "space-around",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "row",
      "msFlexNegative": 0,
      "msFlexPack": "distribute",
      "msFlexPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    onClick={[Function]}
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "auto",
        "WebkitFlexDirection": "column",
        "WebkitFlexShrink": 0,
        "alignItems": "center",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "color": "rgba(210,210,210,1.00)",
        "cursor": "pointer",
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "height": "100%",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "minHeight": "0px",
        "minWidth": "0px",
        "msFlexAlign": "center",
        "msFlexDirection": "column",
        "msFlexNegative": 0,
        "msFlexPreferredSize": "auto",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "textAlign": "center",
        "zIndex": 0,
      }
    }
  >
    <div
      style={
        Object {
          "WebkitAlignItems": "stretch",
          "WebkitBoxAlign": "stretch",
          "WebkitBoxDirection": "normal",
          "WebkitBoxFlex": 2,
          "WebkitBoxOrient": "vertical",
          "WebkitFlexBasis": "0px",
          "WebkitFlexDirection": "column",
          "WebkitFlexGrow": 2,
          "WebkitFlexShrink": 0,
          "alignItems": "stretch",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": "0px",
          "flexDirection": "column",
          "flexGrow": 2,
          "flexShrink": 0,
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "minHeight": "0px",
          "minWidth": "0px",
          "msFlexAlign": "stretch",
          "msFlexDirection": "column",
          "msFlexNegative": 0,
          "msFlexPositive": 2,
          "msFlexPreferredSize": "0px",
          "overflowX": "hidden",
          "overflowY": "hidden",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "position": "relative",
          "width": "40px",
          "zIndex": 0,
        }
      }
    >
      <div
        style={
          Object {
            "WebkitAlignItems": "stretch",
            "WebkitBoxAlign": "stretch",
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "vertical",
            "WebkitFlexBasis": "auto",
            "WebkitFlexDirection": "column",
            "WebkitFlexShrink": 0,
            "alignItems": "stretch",
            "backgroundColor": "rgba(0,0,0,0.00)",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
            "borderBottomStyle": "solid",
            "borderBottomWidth": "0px",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "0px",
            "borderRightStyle": "solid",
            "borderRightWidth": "0px",
            "borderTopStyle": "solid",
            "borderTopWidth": "0px",
            "bottom": "0px",
            "boxSizing": "border-box",
            "display": "flex",
            "flexBasis": "auto",
            "flexDirection": "column",
            "flexShrink": 0,
            "height": "100%",
            "left": "0px",
            "marginBottom": "0px",
            "marginLeft": "0px",
            "marginRight": "0px",
            "marginTop": "0px",
            "minHeight": "0px",
            "minWidth": "0px",
            "msFlexAlign": "stretch",
            "msFlexDirection": "column",
            "msFlexNegative": 0,
            "msFlexPreferredSize": "auto",
            "paddingBottom": "0px",
            "paddingLeft": "0px",
            "paddingRight": "0px",
            "paddingTop": "0px",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "width": "100%",
            "zIndex": -1,
          }
        }
      />
    </div>
    <div
      dir="auto"
      style={
        Object {
          "borderBottomWidth": "0px",
          "borderLeftWidth": "0px",
          "borderRightWidth": "0px",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "color": "inherit",
          "display": "inline",
          "fontFamily": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Ubuntu, \\"Helvetica Neue\\", sans-serif",
          "fontSize": "14px",
          "fontStyle": "inherit",
          "fontVariant": "inherit",
          "fontWeight": "inherit",
          "lineHeight": "inherit",
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "textDecoration": "none",
          "whiteSpace": "pre-wrap",
          "wordWrap": "break-word",
        }
      }
    >
      Rewards
    </div>
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "auto",
        "WebkitFlexDirection": "column",
        "WebkitFlexShrink": 0,
        "alignItems": "center",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "color": "rgba(210,210,210,1.00)",
        "cursor": "pointer",
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "height": "100%",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "minHeight": "0px",
        "minWidth": "0px",
        "msFlexAlign": "center",
        "msFlexDirection": "column",
        "msFlexNegative": 0,
        "msFlexPreferredSize": "auto",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "textAlign": "center",
        "zIndex": 0,
      }
    }
  >
    <div
      style={
        Object {
          "WebkitAlignItems": "stretch",
          "WebkitBoxAlign": "stretch",
          "WebkitBoxDirection": "normal",
          "WebkitBoxFlex": 2,
          "WebkitBoxOrient": "vertical",
          "WebkitFlexBasis": "0px",
          "WebkitFlexDirection": "column",
          "WebkitFlexGrow": 2,
          "WebkitFlexShrink": 0,
          "alignItems": "stretch",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": "0px",
          "flexDirection": "column",
          "flexGrow": 2,
          "flexShrink": 0,
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "minHeight": "0px",
          "minWidth": "0px",
          "msFlexAlign": "stretch",
          "msFlexDirection": "column",
          "msFlexNegative": 0,
          "msFlexPositive": 2,
          "msFlexPreferredSize": "0px",
          "overflowX": "hidden",
          "overflowY": "hidden",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "position": "relative",
          "width": "40px",
          "zIndex": 0,
        }
      }
    >
      <div
        style={
          Object {
            "WebkitAlignItems": "stretch",
            "WebkitBoxAlign": "stretch",
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "vertical",
            "WebkitFlexBasis": "auto",
            "WebkitFlexDirection": "column",
            "WebkitFlexShrink": 0,
            "alignItems": "stretch",
            "backgroundColor": "rgba(0,0,0,0.00)",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
            "borderBottomStyle": "solid",
            "borderBottomWidth": "0px",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "0px",
            "borderRightStyle": "solid",
            "borderRightWidth": "0px",
            "borderTopStyle": "solid",
            "borderTopWidth": "0px",
            "bottom": "0px",
            "boxSizing": "border-box",
            "display": "flex",
            "flexBasis": "auto",
            "flexDirection": "column",
            "flexShrink": 0,
            "height": "100%",
            "left": "0px",
            "marginBottom": "0px",
            "marginLeft": "0px",
            "marginRight": "0px",
            "marginTop": "0px",
            "minHeight": "0px",
            "minWidth": "0px",
            "msFlexAlign": "stretch",
            "msFlexDirection": "column",
            "msFlexNegative": 0,
            "msFlexPreferredSize": "auto",
            "paddingBottom": "0px",
            "paddingLeft": "0px",
            "paddingRight": "0px",
            "paddingTop": "0px",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "width": "100%",
            "zIndex": -1,
          }
        }
      />
    </div>
    <div
      dir="auto"
      style={
        Object {
          "borderBottomWidth": "0px",
          "borderLeftWidth": "0px",
          "borderRightWidth": "0px",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "color": "inherit",
          "display": "inline",
          "fontFamily": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Ubuntu, \\"Helvetica Neue\\", sans-serif",
          "fontSize": "14px",
          "fontStyle": "inherit",
          "fontVariant": "inherit",
          "fontWeight": "inherit",
          "lineHeight": "inherit",
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "textDecoration": "none",
          "whiteSpace": "pre-wrap",
          "wordWrap": "break-word",
        }
      }
    >
      BuySell
    </div>
  </div>
</div>
`;
